### PR TITLE
IEnumerable Except, Intersect overloads

### LINKED ISSUE #50: Refactor out UserStub in favor of using one available from Testing package
Similar to https://github.com/AndcultureCode/AndcultureCode.CSharp.Conductors/issues/26, pull out at least the `UserStub` entity and update any references of tests using it. The latest version(s) (>= 0.6.0) of that package should have a similar entity available with a predefined factory.

### LINKED ISSUE #65: IEnumerable.Intersect and Except overloads with Expression argument
In an attempt to refactor some code, I came across a use case where I wanted to clean up some `Where(...)` and `Any(...)` calls with applicable `Intersect` or `Except` calls. Unfortunately, it looks like the only overloads are for two `IEnumerables` and an optional `IEqualityComparer<T>` - it would be awesome to be able to pass in an expression for comparison similar to how we use `intersect` or `difference` via lodash, etc.

IE instead of:
```cs
var publicationRefProperties = typeof(PublicationReference)
    .GetProperties()
    .Where((e) => !auditablePropertyNames.Contains(e.Name));
```

```cs
var publicationRefProperties = typeof(PublicationReference)
    .GetProperties()
    .Except(auditableProperties, (a, b) => a.Name == b.Name);
```
This stackoverflow answer has an example implementation using `Expression`:

https://stackoverflow.com/a/9526553

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Resolves #65 IEnumerable.Intersect and Except overloads with Expression argument 
Resolves #50 Refactor out UserStub in favor of using one available from Testing package

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [-] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [-] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/AndcultureCode.CSharp.Extensions/AndcultureCode.CSharp.Extensions.md [hunk 1]
```diff
@@ -880,6 +886,18 @@ AndcultureCode.CSharp.Extensions
 
 IEnumerable extension methods
 
+<a name='M-AndcultureCode-CSharp-Extensions-IEnumerableExtensions-Except``1-System-Collections-Generic-IEnumerable{``0},System-Collections-Generic-IEnumerable{``0},System-Func{``0,``0,System-Boolean}-'></a>
+
+### Except\`\`1() `method`
+
+##### Summary
+
+Returns items in source collection that do not exist in second collection based on predicate
```
Comment (jhugs) [inline src/AndcultureCode.CSharp.Extensions/AndcultureCode.CSharp.Extensions.md:895]: Nitpick: Feel free to ignore, but since this is an extension method, the second collection passed is really the first collection passed as it would appear in an invocation. I would maybe reword this to something like "...that do not exist in the given collection based on the predicate" or something

**Review by jhugs — APPROVED** [review]

File: src/AndcultureCode.CSharp.Extensions/AndcultureCode.CSharp.Extensions.md (hunk 1 above)
Comment (jhugs) [reply src/AndcultureCode.CSharp.Extensions/AndcultureCode.CSharp.Extensions.md:895]: Actually, now that I see the parameter is _named_ second, I take this back...

File: src/AndcultureCode.CSharp.Extensions/IEnumerableExtensions.cs [hunk 2]
```diff
@@ -9,6 +10,24 @@ namespace AndcultureCode.CSharp.Extensions
     /// </summary>
     public static class IEnumerableExtensions
     {
+        /// <summary>
+        /// Returns items in source collection that do not exist in second collection based on predicate
+        /// </summary>
+        public static IEnumerable<T> Except<T>(this IEnumerable<T> source, IEnumerable<T> second, Func<T, T, bool> predicate)
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Extensions/IEnumerableExtensions.cs:16]: `second` seems like an odd input name. Maybe something like `excludes` or `excludeItems`?

File: src/AndcultureCode.CSharp.Extensions/IEnumerableExtensions.cs [hunk 3]
```diff
@@ -20,6 +39,19 @@ public static class IEnumerableExtensions
         public static bool HasValues<T>(this IEnumerable<T> source, Func<T, bool> predicate) =>
             !source.IsNullOrEmpty(predicate);
 
+        /// <summary>
+        /// Returns items in source collection that exist in second collection based on the predicate
+        /// </summary>
+        public static IEnumerable<T> Intersect<T>(this IEnumerable<T> source, IEnumerable<T> second, Func<T, T, bool> predicate)
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Extensions/IEnumerableExtensions.cs:45]: Maybe name `second` something like `include` or `includeItems`?

File: src/AndcultureCode.CSharp.Extensions/AndcultureCode.CSharp.Extensions.md (hunk 1 above)
Comment (brandongregoryscott) [reply src/AndcultureCode.CSharp.Extensions/AndcultureCode.CSharp.Extensions.md:895]: I had originally named it `other`, but renamed it based on the `System.Linq` extension method:

```cs
public static IEnumerable<TSource> Except<TSource>(this IEnumerable<TSource> first, IEnumerable<TSource> second, IEqualityComparer<TSource> comparer);
```

I feel like `source` is more appropriate than `first` (and follows the pattern of most other extension methods in here), but I'm not sure if `other` is any better than `second` 🤔

File: test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs [hunk 4]
```diff
@@ -411,15 +426,16 @@ public void OrElse_When_Neither_Expression_Evaluates_As_True_Returns_False()
...
             var janeDoe = Build<UserStub>(
                 UserStubFactory.WITH_YAHOO_EMAIL,
-                (e) => e.Name = $"{JANE}{DOE}"
+                (e) => e.FirstName = JANE,
+                (e) => e.LastName = DOE
             );
             var johnDoe = Build<UserStub>(
                 UserStubFactory.WITH_YAHOO_EMAIL,
-                (e) => e.Name = $"{JOHN}{DOE}"
+                (e) => e.LastName = $"{JOHN}{DOE}"
```
Comment (jstevenson) [inline test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs:438]: Is this correct? If it is, it looks wrong and could use a comment.

File: test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs [hunk 5]
```diff
@@ -460,15 +476,16 @@ public void OrElse_WithNavigationProperty_When_Either_Expression_Evaluates_As_Tr
...
             var janeSmith = Build<UserStub>(
-                (e) => e.Name = $"{JANE}{SMITH}",
+                (e) => e.FirstName = JANE,
+                (e) => e.LastName = SMITH,
                 (e) => e.RelatedUserStub = Build<UserStub>(UserStubFactory.WITH_YAHOO_EMAIL)
             );
             // This entity satisfies the '@gmail.com' email filter but not the 'Smith' name filter
             var johnDoe = Build<UserStub>(
-                (e) => e.Name = $"{JOHN}{DOE}",
+                (e) => e.LastName = $"{JOHN}{DOE}",
```
Comment (jstevenson) [inline test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs:488]: Also here

File: test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs [hunk 6]
```diff
@@ -491,14 +508,15 @@ public void OrElse_WithNavigationProperty_When_Neither_Expression_Evaluates_As_T
...
             // This is the important setup: neither entity satisfies the email or name filter
             var janeDoe = Build<UserStub>(
-                (e) => e.Name = $"{JANE}{DOE}",
+                (e) => e.FirstName = JANE,
+                (e) => e.LastName = DOE,
                 (e) => e.RelatedUserStub = Build<UserStub>(UserStubFactory.WITH_YAHOO_EMAIL)
             );
             var johnDoe = Build<UserStub>(
-                (e) => e.Name = $"{JOHN}{DOE}",
+                (e) => e.LastName = $"{JOHN}{DOE}",
```
Comment (jstevenson) [inline test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs:519]: And finally here

**Review by jstevenson — APPROVED** [review]

File: test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs (hunk 4 above)
Comment (brandongregoryscott) [reply test/AndcultureCode.CSharp.Extensions.Tests/ExpressionExtensionsTest.cs:438]: Ehhh, it's not "wrong" per se - more of an unnecessary piece of setup at this point. When the `UserStub` was ported to the testing package, its fields slightly changed. These tests could probably drop the `FirstName` setup altogether, although the variable names might be harder to come up with. I'll take another pass.

**Comment by brandongregoryscott** [comment]
@all-contributors add @jhugs for review

**Comment by brandongregoryscott** [comment]
@all-contributors add @jstevenson for review
